### PR TITLE
Build release and remove remnants of dotnetty

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: authentication/src/Logging.cs [hunk 1]
```diff
@@ -4,19 +4,17 @@
...
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
 using System.Diagnostics.Tracing;
 using System.Globalization;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
 namespace Microsoft.Azure.Devices
 {
-    [SuppressMessage("Microsoft.Naming", "CA1724:TypeNamesShouldNotMatchNamespaces", Justification = "Conflicts with DotNetty.Common.Internal.Logging")]
```
Comment (drwill-ms) [inline authentication/src/Logging.cs:15]: We don't need this any longer now that we've dropped our dotnetty reference.

File: iothub/device/samples/getting started/FileUploadSample/FileUploadSample.csproj [hunk 2]
```diff
@@ -16,9 +16,6 @@
     <PackageReference Include="Azure.Storage.Blobs" Version="12.13.0" />
     <PackageReference Include="CommandLineParser" Version="2.9.1" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
-
-    <!-- Working around a bug between DotNetty and .NET 5.0+ by specifying the exact version of DotNetty.Common that has the fix -->
```
Comment (drwill-ms) [inline iothub/device/samples/getting started/FileUploadSample/FileUploadSample.csproj:20]: Same, remnant reference.

File: iothub/device/samples/getting started/FileUploadSample/FileUploadSample.csproj [hunk 3]
```diff
@@ -27,11 +24,12 @@
     </None>
   </ItemGroup>
 
-  <ItemGroup Condition="'$(Configuration)' == 'Release'">
+  <!--<ItemGroup Condition="'$(Configuration)' == 'Release'">
```
Comment (drwill-ms) [inline iothub/device/samples/getting started/FileUploadSample/FileUploadSample.csproj:27]: Until we're near end of previews or even in GA, we'll need to delay this strategy.

**Review by timtay-microsoft — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**